packages/sdk/src/swap/v2/types.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  AffiliateBroker,
+  FillOrKillParams,
+  SwapFee,
+  type CcmParams,
+  type DcaParams as Dca,
+} from '@/shared/schemas';
+import { SwapType } from '@/swap/client';
+import { FailureMode } from '@/swap/utils/swap';
+import { ChainsAndAssets } from '../types';
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace SwapV2 {
+  type DcaParameters = Omit<Dca, 'chunkInterval'> & {
+    chunkIntervalBlocks: string;
+  };
+
+  type Ccm = CcmParams & {
+    ccmDepositReceivedBlockIndex: string | undefined;
+  };
+  type Failure = {
+    failedAt: number;
+    failedAtBlockIndex: string;
+    mode: FailureMode;
+    reason: {
+      code: string;
+      message: string;
+    };
+  };
+
+  type WithFailure<T> = T & { failure: Failure | undefined };
+
+  type Boost = {
+    maxBoostFeeBps: number;
+    effectiveBoostFeeBps: number | undefined;
+    boostedAt: number | undefined;
+    boostedBlockIndex: string | undefined;
+    skippedAt: number | undefined;
+    skippedBlockIndex: string | undefined;
+  };
+  type DepositChannelFields = {
+    createdAt: number;
+    brokerCommissionBps: number;
+    depositAddress: string;
+    srcChainExpiryBlock: string;
+    estimatedExpiryTime: number;
+    expectedDepositAmount: string;
+    isExpired: boolean;
+    openedThroughBackend: boolean;
+    affiliateBrokers: AffiliateBroker[];
+    fillOrKillParams: FillOrKillParams | undefined;
+    dcaParams: DcaParameters | undefined;
+    srcChainRequiredBlockConfirmations: number | null | undefined;
+  };
+
+  type DepositFields = {
+    amount: string | undefined;
+    txRef: string | undefined;
+    txConfirmations: number | undefined;
+    receivedAt: number | undefined;
+    receivedBlockIndex: string | undefined;
+  };
+
+  type Swap = {
+    swapInputAmount: string;
+    swapOutputAmount: string | undefined;
+    scheduledAt: number;
+    scheduledBlockIndex: string;
+    executedAt: number | undefined;
+    executedBlockIndex: string | undefined;
+    retryCount: number;
+    latestSwapRescheduledAt: number | undefined;
+    latestSwapRescheduledBlockIndex: number | undefined;
+  };
+
+  type SwapFields = {
+    totalInputAmountSwapped: string | undefined;
+    totalOutputAmountSwapped: string | undefined;
+    totalChunksExecuted: number;
+    isDcaSwap: boolean;
+    lastExecutedChunk: Swap | undefined;
+    currentChunk: Swap;
+    type: SwapType;
+    fees: SwapFee[];
+  };
+
+  interface SwapStatusResponseCommonFields extends ChainsAndAssets {
+    swapId: string;
+    destAddress: string;
+    depositChannel: DepositChannelFields;
+    ccm: Ccm | undefined;
+    boost: Boost | undefined;
+    estimatedDurationSeconds: number | null | undefined;
+  }
+
+  type EgressFields = {
+    amount: string;
+    scheduledAt: number;
+    scheduledBlockIndex: string;
+  };
+  type EgressSentFields = EgressFields & {
+    sentTxRef: string;
+  };
+  type EgressCompletedFields = EgressFields & {
```

Comment:
while it is correct to have different types for a scheduled and completed egress, i am not sure if its worth it. i would be fine with just having one type:
```
  type EgressFields = {
    amount: string;
    scheduledAt: number;
    scheduledBlockIndex: string;
    sentTxRef: string | undefined;
    sentAt: number | undefined;
    sentAtBlockIndex: string | undefined;
  };
```